Documentation/Wiki/Advanced-Features/Graph-Reuse.md

```diff
@@ -0,0 +1,119 @@
+for checking the pip graph inputs, i.e., besides tracking the content hashes of the evaluated spec files, BuildXL also keeps track of the USNs of those files.
+
+If the pip graph in the engine cache cannot be reused, then BuildXL will consult to the real cache. The pip graph weak fingerprint and the hash of the pip graph inputs
+become the key of the cache look up.
+
+# Evaluation filter
+
+Users may pass different evaluation filters when constructing the pip graph. The evaluation filters only make the constructed pip graph a subgraph of the one that is constructed
+without any evaluation filter. Thus, the latter graph is actually reusable when the evaluation filters are applied.
+
+# Making pip graph caching work
+
+To have pip graph cache hit, users need to make the pip graph weak fingerprint and the pip graph inputs stable build over build. Here are some guidelines to make the graph inputs stable.
+
+## Do not query constantly changing environment variable
+
+Typical case of using constantly changing environment variable is using an environment variable to keep track of build id. For example,
+```
+const buildId = Environment.getStringValue("BUILD_ID"); 
+```
+where `%BUILD_ID%` is set by a batch script before that script invokes a build, e.g., 
+```
+# REM Get build id based on the current time.
+set BUILD_ID=...
+
+# Invoke build
+bxl /c:config.dsc ...
+``` 
+
+## Use BuildXL's log folder mount for current build's log
+
+Related to the constantly changing environment variable, users typically invent their own log folders based on the current time. For example,
+```
+Transformer.execute({
+    ...
+    outputs: [p`${Environment.getPathValue("LOG_PATH")}/pip123.log`]
+    ...
+});
+```
+where `%LOG_PATH%` is set by a batch script before that script invokes a build, e.g., 
+```
+# REM Get log path based on the current time.
+set LOG_PATH=...
+
+# Invoke build
+bxl /c:config.dsc ...
+```
+
+Instead of using user-specified log folders, use BuildXL's mount, e.g.,
+```
+Transformer.execute({
+    ...
+    outputs: [p`${Context.getMount("LogsDirectory")}/pip123.log`]
+    ...
+});
+```
+
+## Enable user-profile redirection
+
+For graph caching across machines, do not query or embed user profile or domain in the pip specification without enabling user-profile redirection.
+For example, 
+```
+Transformer.execute({
+    ...
+    untrackedPaths: [p`${Environment.getPathValue("USERPROFILE")}/abc.txt`]
+    ...
+})
+```
+The value of `Environment.getPathValue("USERPROFILE")` will be different from one user to another.
+
+BuildXL provides a so-called user-profile redirection that will redirect the values of `%USERPROFILE%` and related environment variables
+to stable values. To this end, BuildXL creates a junction from a stable path to the real user profile path. To enable user-profile
+redirection, see [user profile redirection](ToDo.md)
```

Comment:
Is this enabled by default?

Reply:
User redirection is only enabled by default when building in CB.